common/browser_context.go

```diff
@@ -546,6 +547,131 @@ func (b *BrowserContext) Cookies(urls ...string) ([]*api.Cookie, error) { //noli
 			SameSite: api.CookieSameSite(c.SameSite),
 		}
 	}
+	// filter cookies by the provided URLs.
+	cookies, err = filterCookies(cookies, urls...)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving cookies: %w", err)
```

Comment:
How about:

```suggestion
		return nil, fmt.Errorf("filtering cookies: %w", err)
```

Just so that it's different from the error above and so possibly easier to debug?